### PR TITLE
Update UndoSequenceDiagram-Logic & -Model uml

### PR DESCRIPTION
**Pull request by JerryWang0000**
Replace AddressBook with StaffBook, AddressBookParser with StaffConnectParser.

### CODE REVIEW

**Comment by JerryWang0000** [comment]
I can also update UndoRedoState umls actually.

File: docs/diagrams/UndoSequenceDiagram-Logic.puml [hunk 1]
```diff
@@ -4,7 +4,7 @@ skinparam ArrowFontStyle plain
 
 box Logic LOGIC_COLOR_T1
 participant ":LogicManager" as LogicManager LOGIC_COLOR
-participant ":AddressBookParser" as AddressBookParser LOGIC_COLOR
+participant ":StaffBookParser" as StaffBookParser LOGIC_COLOR
```
Comment (tsulim) [inline docs/diagrams/UndoSequenceDiagram-Logic.puml]: I think you might have mixed this one up. `StaffConnectParser`

**Review by tsulim — COMMENTED** [review]
Overall, it looks good. However, I noticed that you made a typo on one of the names.

File: docs/diagrams/UndoSequenceDiagram-Logic.puml (hunk 1 above)
Comment (JerryWang0000) [reply docs/diagrams/UndoSequenceDiagram-Logic.puml]: Indeed a Typo

**Review by tsulim — APPROVED** [review]
LGTM! 👍

**Pull request merged**